Cubical/Categories/Constructions/Hyperdoctrine.agda

```diff
@@ -0,0 +1,63 @@
+{-# OPTIONS --safe #-}
+
+module Cubical.Categories.Constructions.Hyperdoctrine where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Categories.Category
+open import Cubical.Categories.Functor.Base
+open import Cubical.Categories.NaturalTransformation.Base
+
+open import Cubical.Categories.Instances.PosetCat.Base
+open import Cubical.Categories.Instances.PosetCat.Monotone
+open import Cubical.Categories.Instances.PosetCat.MonotoneAdjoint
+open import UMP
```

Comment:
I think we should maybe deprecate this module. I would instead use a category with chosen binary products

Reply:
Refactored to use the Limit.BinProduct with a field to assume the binproducts exist.